homeassistant/components/homekit/type_covers.py

```diff
@@ -83,20 +95,31 @@ def __init__(self, *args):
     @debounce
     def move_cover(self, value):
         """Move cover to value if call came from HomeKit."""
+        self.hass.add_job(self.async_move_cover, value)
+
+    async def async_move_cover(self, value):
+        """Move cover to value if call came from HomeKit. Coroutine."""
         _LOGGER.debug('%s: Set position to %d', self.entity_id, value)
         self.homekit_target = value
 
-        params = {ATTR_ENTITY_ID: self.entity_id, ATTR_POSITION: value}
-        self.hass.services.call(DOMAIN, SERVICE_SET_COVER_POSITION, params)
+        params = {ATTR_ENTITY_ID: self.entity_id,
+                  ATTR_POSITION: value}
+        await self.hass.services.async_call(
+            DOMAIN, SERVICE_SET_COVER_POSITION, params)
 
-    def update_state(self, new_state):
-        """Update cover position after state changed."""
+    def async_update_state(self, new_state):
+        """Update cover position after state changed.
+
+        Method is run in the event loop.
+        """
         current_position = new_state.attributes.get(ATTR_CURRENT_POSITION)
         if isinstance(current_position, int):
-            self.char_current_position.set_value(current_position)
+            self.hass.async_add_job(
```

Comment:
You need to await these calls if you want to mimic the old behavior (and thus method needs to be async). However, If you're going to call to the thread pool twice, you might aswell run the whole method in a thread right away, probably more efficient.